interface/frontend/src/components/open-image/ImageSeries.vue

```diff
@@ -97,54 +64,52 @@
           type: 'DICOM',
           prefixCS: '://',
           prefixUrl: '/api/images/metadata?dicom_location=/',
-          paths: []
+          paths: null
         },
-        selected: null,
-        showImport: false
+        currentCase: null
       }
     },
     created () {
       this.fetchData()
       this.fetchAvailableImages()
     },
     mounted: function () {
-      EventBus.$on('dicom-selection', (path) => {
-        this.preview.paths = path
-        console.log(this.preview)
+      EventBus.$on('dicom-selection', (paths) => {
+        this.preview.paths = paths
+      })
+
+      EventBus.$on('start-new-case', (filePath) => {
```

Comment:
I wonder why is this using EventBus? The method to send new case can be invoked from startNewCase directly. If the params filePath is missing, we can simply pass it into this ImageSeries component via the props down in the TreeViews.

I think we should use EventBus sparingly to ensure components are modular and feature-compact. (That is, if I take this component and use it somewhere else, I don't have to worry about emitting an event from the parent component, but I can depend on this component to handle the job)

Reply:
@louisgv, I am new to Vue.js so may be wrong but it seems like there is no simple way to implement two way binding between one component and it's ground child. We can emit events and catch+emit them on each middleman, but it doesn't seem nice to me. Also we can use DOM method `this.$el.dispatchEvent` like suggested [here](https://stackoverflow.com/questions/42615445/vuejs-2-0-emit-event-from-grand-child-to-his-grand-parent-component).

Reply:
@Serhiy-Shekhovtsov I think I should have commented on the function down there instead of here :p 

My thought was, is there a way to make the grand-child component feature complete without the need to rely on the grand-parent component? Since if the grand-child needs anything, the grand-parent can simply pass it down. Thus, we can maintain the uni-directional data flow, and will not have to worry about propagating data upstream, and will also ensure the grand-child has all the feature it need to be stand-alone and re-usable.